features/step_definitions/your_contributions_steps.rb

```diff
@@ -2,22 +2,35 @@
   step 'I am on step 1 of the WPCC homepage'
 end
 
-Given(/^I complete the first step and move on to second step$/) do
+Given(/^I complete the your details form and move to your contributions page$/) do
   step 'I fill in the age, gender, salary and frequency fields'
   step 'I click on "My employer makes contributions on part of my salary"'
   step 'I click the Next button'
 end
 
-When(/^I click edit on the first step section$/) do
-  find_link('edit').click
+Given(/^I complete the your details and your contributions forms and move to your results page$/) do
+  step 'I fill in the age, gender, salary and frequency fields'
+  step 'I click on "My employer makes contributions on part of my salary"'
+  step 'I click the Next button'
+  save_and_open_page
+  click_button
+end
+
+When(/^I click edit on the your details summary$/) do
+  sections = page.find_all('.section--details')
+  within(sections.first) do
+    find_link('edit').click
```

Comment:
I know we've had this conversation before, but I dont think its good practice to add class names solely for the purpose of testing. The extra code in the tests wont affect performance, the extra code in the markup will. If the test fails due to a change in markup, it should be addressed at that point.